pandas/tests/frame/test_constructors.py

```diff
@@ -283,6 +283,11 @@ def test_constructor_cast_failure(self):
         with pytest.raises(ValueError, match=msg):
             df["test"] = np.ones((4, 2))
 
+        # this is not ok
+        msg = "Expected a 1D array, got an array with shape \\(4, 2, 3\\)"
```

Comment:
this file is for constructor tests.  can you put a dedicated test in tests/frame/indexing/test_setitem.py